server/src/main/java/org/candlepin/dto/api/v1/AsyncJobStatusTranslator.java

```diff
@@ -58,27 +60,45 @@ public AsyncJobStatusDTO populate(AsyncJobStatus source, AsyncJobStatusDTO desti
     public AsyncJobStatusDTO populate(ModelTranslator translator, AsyncJobStatus source,
         AsyncJobStatusDTO destination) {
 
-        destination = super.populate(translator, source, destination);
+        if (source == null) {
+            throw new IllegalArgumentException("source is null");
+        }
+
+        if (destination == null) {
+            throw new IllegalArgumentException("destination is null");
+        }
+
+        Date created = source.getCreated();
+        destination.setCreated(created != null ? created.toInstant().atOffset(ZoneOffset.UTC) : null);
+
+        Date updated = source.getUpdated();
+        destination.setUpdated(updated != null ? updated.toInstant().atOffset(ZoneOffset.UTC) : null);
 
         destination.setId(source.getId());
-        destination.setJobKey(source.getJobKey());
+        destination.setKey(source.getJobKey());
         destination.setName(source.getName());
         destination.setGroup(source.getGroup());
         destination.setOrigin(source.getOrigin());
         destination.setExecutor(source.getExecutor());
         destination.setPrincipal(source.getPrincipalName());
-        destination.setStartTime(source.getStartTime());
-        destination.setEndTime(source.getEndTime());
-        destination.setAttempts(source.getAttempts());
-        destination.setMaxAttempts(source.getMaxAttempts());
-        destination.setResult(source.getJobResult());
 
         JobState state = source.getState();
         destination.setState(state != null ? state.name() : null);
 
         JobState pstate = source.getPreviousState();
         destination.setPreviousState(pstate != null ? pstate.name() : null);
 
+        Date startTime = source.getStartTime();
+        destination.setStartTime(startTime != null ? startTime.toInstant().atOffset(ZoneOffset.UTC) : null);
+
+        Date endTime = source.getEndTime();
+        destination.setEndTime(endTime != null ? endTime.toInstant().atOffset(ZoneOffset.UTC) : null);
+
+        destination.setAttempts(source.getAttempts());
+        destination.setMaxAttempts(source.getMaxAttempts());
+        destination.setStatusPath("/jobs/" + source.getId());
```

Comment:
We should be more cautious here, checking for null, and using format instead of concatenation: `source.getId() != null ? String.format("/jobs/%s", source.getId()) : null`

Reply:
agreed, corrected.